src/plz/plz.go

```diff
@@ -38,20 +38,31 @@ func Run(targets, preTargets []core.BuildLabel, state *core.BuildState, config *
 	// Start looking for the initial targets to kick the build off
 	go findOriginalTasks(state, preTargets, targets, arch)
 
-	parses, builds, tests, remoteBuilds, remoteTests := state.TaskQueues()
+	parses, builds, remoteBuilds, tests, remoteTests := state.TaskQueues()
 
 	// Start up all the build workers
 	var wg sync.WaitGroup
-	wg.Add(config.Please.NumThreads + config.NumRemoteExecutors())
+	wg.Add(1 + config.Please.NumThreads + config.NumRemoteExecutors())
+	go func() {
+		// All parses happen async on separate goroutines so we don't have to worry about them blocking.
+		// They manage concurrency control themselves.
+		for task := range parses {
+			go func(task core.ParseTask) {
+				parse.Parse(0, state, task.Label, task.Dependent, task.ForSubinclude)
+				state.TaskDone()
+			}(task)
+		}
+		wg.Done()
+	}()
```

Comment:
nit: Took me a little while to figure out what all the `go ...` things were doing here. Might read better if this was:

```
go doParses(parses)
```

Reply:
I have a bit of an aversion to passing the waitgroup out to another function...